### PR TITLE
fix creation of nested types

### LINKED ISSUE #64: create nested types
It appears nested types are not created. 

datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

``` json
    "ExecuteDataFlowActivityTypeProperties": {
      "description": "Execute data flow activity properties.",
      "properties": {
        "dataFlow": {
          "description": "Data flow reference.",
          "$ref": "../datafactory.json#/definitions/DataFlowReference"
        },
        "staging": {
          "description": "Staging info for execute data flow activity.",
          "$ref": "../datafactory.json#/definitions/DataFlowStagingInfo"
        },
        "integrationRuntime": {
          "description": "The integration runtime reference.",
          "$ref": "../datafactory.json#/definitions/IntegrationRuntimeReference"
        },
        "compute": {
          "description": "Compute properties for data flow activity.",
          "type": "object",
          "properties": {
            "computeType": {
              "description": "Compute type of the cluster which will execute data flow job.",
              "type": "string",
              "enum": [
                "General",
                "MemoryOptimized",
                "ComputeOptimized"
              ],
              "x-ms-enum": {
                "name": "DataFlowComputeType",
                "modelAsString": true
              }
            },
            "coreCount": {
              "description": "Core count of the cluster which will execute data flow job. Supported values are: 8, 16, 32, 48, 80, 144 and 272.",
              "type": "integer"
            }
          }
        }
      },
```

``` rust
mod execute_data_flow_activity_type_properties {
    use super::*;
    #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
    pub struct Compute {
        #[serde(rename = "computeType", skip_serializing_if = "Option::is_none")]
        pub compute_type: Option<compute::ComputeType>,
        #[serde(rename = "coreCount", skip_serializing_if = "Option::is_none")]
        pub core_count: Option<i64>,
    }
}
```

`compute::ComputeType` is not being created.

### PR DESCRIPTION
**Pull request by ctaggart**
fix #64

### CODE REVIEW

File: codegen/src/codegen.rs [hunk 1]
```diff
@@ -281,23 +278,24 @@ impl CodeGen {
...
             None => {
                 if is_local_enum(property) {
                     let (tp_name, tp) = create_enum(namespace, property_name, property);
-                    Ok((tp_name, Some(tp)))
+                    Ok((tp_name, vec![tp]))
                 } else if is_local_struct(property) {
                     let id = ident(&property_name.to_camel_case());
                     let tp_name = quote! {#namespace::#id};
                     let tps = self.create_struct(doc_file, property_name, property)?;
-                    Ok((tp_name, Some(tps[0].clone())))
```
Comment (ctaggart) [inline codegen/src/codegen.rs:298]: Here is the bug. I was just getting the first type, which left off the mod that was created.

**Review by rylev — APPROVED** [review]
I'm finding the codegen code a bit hard to follow. Perhaps we can leave a few more doc comments to make the flow a bit clear. I find that giving an example of what each code gen method is generating helps a lot.

**Comment by ctaggart** [comment]
@rylev, yes, while fixing stuff, I'll see able adding more doc comments with examples.
